back-end/e-learning/app/Models/User.php

```diff
@@ -57,4 +58,32 @@ public function getUserCategory($id)
     {
         return $this->userCategories()->where('category_id', $id)->first();
     }
+
+    public function followings()
+    {
+        return $this->belongsToMany(User::class, 'followers', 'user_id', 'user_following_id');
+    }
+    
+    public function followers()
+    {
+        return $this->belongsToMany(User::class, 'followers', 'user_following_id', 'user_id');
+    }
+
+    public function isFollowing($id)
+    {
+        $following = $this->followings()->where('user_following_id', $id)->first();
+        if (is_null($following))
+        {
+            return false;
+        }
+        else
+        {
+            return true;
+        }
```

Comment:
```suggestion
       return !is_null($following)
```

I think this can be shortened to above na lang